src/main/java/org/jboss/xavier/integrations/route/VMWorkloadInventoryRoutes.java

```diff
@@ -28,8 +30,23 @@ public void configure() throws Exception {
         from("direct:calculate-vmworkloadinventory").routeId("calculate-vmworkloadinventory")
             .setHeader("KieSessionId", constant("WorkloadInventoryKSession0"))
             .bean("VMWorkloadInventoryCalculator", "calculate(${body}, ${header.${type:org.jboss.xavier.integrations.route.MainRouteBuilder.MA_METADATA}})", false)
-            .split(body()).parallelProcessing(parallel).aggregationStrategy(new WorkloadInventoryReportModelAggregationStrategy())
+                .process(exchange -> {
+                    Set<String> vmNamesWithSharedDisk = exchange.getIn().getHeader("vmNamesWithSharedDisk", Set.class);
+                    Collection<VMWorkloadInventoryModel>  allVms = exchange.getIn().getBody(Collection.class);
+                    Collection<VMWorkloadInventoryModel> originalAndAmendedVms = allVms.stream().map(vmWorkloadInventoryModel ->
+                    {
+                        if (vmNamesWithSharedDisk != null && vmNamesWithSharedDisk.contains(vmWorkloadInventoryModel.getVmName())) {
+                            vmWorkloadInventoryModel.setHasSharedVmdk(true);
+                        }
+                        return vmWorkloadInventoryModel;
+                    }).collect(Collectors.toList());
```

Comment:
```suggestion
                    allVms.forEach(vmWorkloadInventoryModel ->
                    {
                        if (vmNamesWithSharedDisk != null && vmNamesWithSharedDisk.contains(vmWorkloadInventoryModel.getVmName())) {
                            vmWorkloadInventoryModel.setHasSharedVmdk(true);
                        }
                    });
```

@m-brophy this just a suggestion, it seems to me that `originalAndAmendedVms` is not used so better to use a `forEach()` instead of `map()` since we only need to iterate the collection `allVms`.